src/client/eppo-client.spec.ts

```diff
@@ -369,6 +369,229 @@ describe('EppoClient E2E test', () => {
     expect(assignment).toEqual('control');
   });
 
+  describe('assignment logging deduplication', () => {
+    it('logs duplicate assignments without an assignment cache', () => {
+      const mockLogger = td.object<IAssignmentLogger>();
+
+      storage.setEntries({ [flagKey]: mockExperimentConfig });
+      const client = new EppoClient(storage);
+      client.setLogger(mockLogger);
```

Comment:
This could be folded into a `beforeAll()` (if you can reset testdouble spy) or `beforeEach()` block, cleaning up duplication in the other test cases.